tests/benchmark_expr_templates.cpp

```diff
@@ -178,12 +201,38 @@ static void BM_large_expr_single_alloc_par_EXPR(benchmark::State& state)
   MPI_Barrier(MPI_COMM_WORLD);
 }
 
+static void BM_large_expr_single_alloc_hypre_par_EXPR(benchmark::State& state)
+{
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  // Number of rows is the argument that varies
+  const int rows  = state.range(0);
+  auto [lhs, rhs] = sample_vectors(rows);
+
+  auto [partitioning, start] = build_partitioning(MPI_COMM_WORLD, rows);
+
+  mfem::HypreParVector lhs_par(MPI_COMM_WORLD, rows, lhs + start, partitioning.get());
+  mfem::HypreParVector rhs_par(MPI_COMM_WORLD, rows, rhs + start, partitioning.get());
+
+  mfem::HypreParVector expr_result(MPI_COMM_WORLD, rows, partitioning.get());
+
+  for (auto _ : state) {
+    // This code gets timed
+    evaluate(lhs + rhs + lhs + rhs + lhs + rhs + lhs + rhs + lhs + rhs + lhs + rhs + lhs + rhs, expr_result);
+  }
+
+  MPI_Barrier(MPI_COMM_WORLD);
+}
+
 BENCHMARK(BM_mixed_expr_MFEM)->RangeMultiplier(2)->Range(10, 10 << 10);
 BENCHMARK(BM_mixed_expr_EXPR)->RangeMultiplier(2)->Range(10, 10 << 10);
 BENCHMARK(BM_mixed_expr_single_alloc_EXPR)->RangeMultiplier(2)->Range(10, 10 << 10);
 BENCHMARK(BM_large_expr_MFEM)->RangeMultiplier(2)->Range(10, 10 << 10);
 BENCHMARK(BM_large_expr_single_alloc_EXPR)->RangeMultiplier(2)->Range(10, 10 << 10);
-BENCHMARK(BM_large_expr_single_alloc_par_EXPR)->RangeMultiplier(2)->Range(10, 10 << 10);
+
+// Too slow
```

Comment:
If we changed the large expr examples to use sparse matrices instead of dense, it could reduce the runtime of those tests and would be more representative of our calculations too.